analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.OperationValue.cs

```diff
@@ -125,6 +125,19 @@ public void SetOperationValue_WithWrapper_Overwrites()
             sut[operation].Should().Be(value2);
         }
 
+        [TestMethod]
+        public void SetOperationValue_NullValue_RemovesSymbol()
+        {
+            var value1 = new SymbolicValue(new SymbolicValueCounter());
+            var operation = new IOperationWrapperSonar(TestHelper.CompileCfgBodyCS("var x = 42;").Blocks[1].Operations[0]);
+            var sut = ProgramState.Empty;
+
+            sut[operation].Should().BeNull();
+            sut = sut.SetOperationValue(operation, value1);
```

Comment:
We don't care if empty was empty. We should check that value was added before we remove it.
```suggestion
                        sut = sut.SetOperationValue(operation, value1);
                        sut[operation].Should().NotBeNull();
```